src/ploomber/cli/task.py

```diff
@@ -59,10 +62,15 @@ def _task_cli(accept_task_id=False):
     if no_flags or args.build:
         try:
             task.build(force=args.force)
-        except Exception:
+        except Exception as e:
             if getattr(args, 'task_id', None):
                 api.tasks_update(getattr(args, 'task_id'), 'failed')
-            raise
+
+            exception_string = task_build_exception(task=task,
+                                                    message=_format
+                                                    .exception(e),
+                                                    exception=e)
+            raise TaskBuildError(exception_string)
```

Comment:
I think we'll need to do a bit of experimentation.

I think we can get rid of `task_build_exception`.

some quick idea (we'll probably need to iterate on this but just to see what happens).

let's try something like:

```python
import sys

try:
    task.build(force=args.force)
except Exception as e:
    print(_format.exception(e))
    sys.exit(1)
```

I think this should make the error more concise

Reply:
@grnnja any update on this?

Reply:
I'll look into this

Reply:
I tried the change but the behavior is the same as it was without formatting. The output is more concise than by passing it through `task_build_exception` but it is not fancily formatted:
```bash
(ploomber) conda-shell-chrootenv:prem@morokei:~/Documents/projects/ploomber/ploomberw/test/example$ ploomber task fit
Loading pipeline...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 17403.75it/s]
Traceback (most recent call last):
  File "/home/prem/.local/lib/python3.9/site-packages/ploomber/tasks/abc.py", line 554, in _build
    res = self._run()
  File "/home/prem/.local/lib/python3.9/site-packages/ploomber/tasks/abc.py", line 638, in _run
    raise TaskBuildError('Attempted to run task "{}", whose '
ploomber.exceptions.TaskBuildError: Attempted to run task "fit", whose status TaskStatus.Skipped. Render again and set force=True if you want to force execution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/prem/.local/lib/python3.9/site-packages/ploomber/cli/io.py", line 52, in wrapper
    fn(**kwargs)
  File "/home/prem/.local/lib/python3.9/site-packages/ploomber/telemetry/telemetry.py", line 500, in wrapper
    raise e
  File "/home/prem/.local/lib/python3.9/site-packages/ploomber/telemetry/telemetry.py", line 488, in wrapper
    result = func(*args, **kwargs)
  File "/home/prem/.local/lib/python3.9/site-packages/ploomber/cli/task.py", line 76, in main
    _task_cli()
  File "/home/prem/.local/lib/python3.9/site-packages/ploomber/cli/task.py", line 61, in _task_cli
    task.build(force=args.force)
  File "/home/prem/.local/lib/python3.9/site-packages/ploomber/tasks/abc.py", line 522, in build
    res, _ = self._build(catch_exceptions=catch_exceptions)
  File "/home/prem/.local/lib/python3.9/site-packages/ploomber/tasks/abc.py", line 573, in _build
    raise TaskBuildError(msg) from e
ploomber.exceptions.TaskBuildError: Error building task "fit"

ploomber.exceptions.TaskBuildError: Attempted to run task "fit", whose status TaskStatus.Skipped. Render again and set force=True if you want to force execution
ploomber.exceptions.TaskBuildError: Error building task "fit"
```

I made `task_build_exception` since I think `ploomber build`'s exceptions run through the classes in [messagecollectory.py](https://github.com/ploomber/ploomber/blob/846f5bab6956ce7ef04e7384b2d038555a22759e/src/ploomber/messagecollector.py) and some of the text formatting is handled in there. In particular the equal signs for the heading:
```bash
============================== Task build failed ===============================
------------------ DownloadFromURL: download -> File('file') -------------------
hello
=============================== Summary (1 task) ===============================
DownloadFromURL: download -> File('file')
============================== Task build failed ===============================
```
Maybe we don't this though? I'm not really sure

Reply:
I tried the suggested change and it is shorter. but it doesn't have the fancy formatting with the equal signs
```bash
============================== Task build failed ===============================
------------------ DownloadFromURL: download -> File('file') -------------------
hello
=============================== Summary (1 task) ===============================
DownloadFromURL: download -> File('file')
============================== Task build failed ===============================
```

I made `task_build_exception` since I think `ploomber build`'s exceptions run through the classes in [messagecollectory.py](https://github.com/ploomber/ploomber/blob/846f5bab6956ce7ef04e7384b2d038555a22759e/src/ploomber/messagecollector.py) and some of the text formatting is handled in there.
Maybe we don't this though? I'm not really sure

Reply:
so the example you showed has the equal signs. can you show how your implementation looks like?

Reply:
Yeah, it's using `task_build_exception` which uses [code in MessageCollector](https://github.com/ploomber/ploomber/blob/846f5bab6956ce7ef04e7384b2d038555a22759e/src/ploomber/messagecollector.py) for formatting

Specifically the example I have is from this test I have in the pr:
```python

    dag = DAG()

    url = """
    https://google.com
    """
    task = DownloadFromURL(url, File('file'), dag=dag, name='download')

    exceptionText = mc.task_build_exception(task=task,
                                            message="hello",
                                            exception=Exception("hello)"))
 ```